dlt/destinations/impl/databricks/databricks_adapter.py

```diff
@@ -53,21 +67,78 @@ def databricks_adapter(
         >>> data = [{"name": "Marcel", "description": "Raccoon Engineer", "date_hired": 1700784000}]
         >>> databricks_adapter(data, cluster="date_hired", table_comment="Employee Data",
         ...     table_tags=["production", {"environment": "prod"}, "employees"])
+        >>> # Use AUTO clustering
+        >>> databricks_adapter(data, cluster="AUTO", table_comment="Auto-clustered table")
+        >>> # Use partitioning
+        >>> databricks_adapter(data, partition=["year", "month"], cluster="customer_id")
+        >>> # Create Iceberg table
+        >>> databricks_adapter(data, table_format="ICEBERG", cluster="customer_id")
     """
     resource = get_resource_for_adapter(data)
 
     additional_table_hints: Dict[str, TTableHintTemplate[Any]] = {}
     additional_column_hints: TDatabricksTableSchemaColumns = {}
 
+    # Handle table format
+    if table_format not in ["DELTA", "ICEBERG"]:
+        raise ValueError("`table_format` must be either 'DELTA' or 'ICEBERG'.")
+
+    # Store table format as a table hint
+    additional_table_hints[TABLE_FORMAT_HINT] = table_format
+
+    # Validate Iceberg-specific constraints
+    if table_format == "ICEBERG":
```

Comment:
Oh OK, this you'd need to do when creating table... because `table_format` is not available at this point. and convert error to warnings (that given prop has no effect on ICEBERG) so we do not fail the load

other alternative is to implement
```py
def verify_schema(
        self, only_tables: Iterable[str] = None, new_jobs: Iterable[ParsedLoadJobFileName] = None
    ) -> List[PreparedTableSchema]:
```
on `DatabricksClient` but this check will happen in load phase so the user experience is still not as good as failing early or ignoring with warning.